docs/getting_started/code_samples.rst

```diff
@@ -1,42 +1,81 @@
+- `Example 4: SetupRayCluster, SubmitRayJob & DeleteRayCluster`_
 
-Scenario 1: Setting up a Ray cluster on an existing Kubernetes cluster
-----------------------------------------------------------------------
+Example 1: Ray jobs on an existing cluster
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+If you already have a Ray cluster set up, you can use the ``SubmitRayJob`` operator or ``ray.task()`` decorator to submit jobs directly.
 
-If you have an existing Kubernetes cluster and want to install a Ray cluster on it, and then run a Ray job, you can use the ``SetupRayCluster``, ``SubmitRayJob``, and ``DeleteRayCluster`` operators.
+In the example below (``ray_taskflow_example_existing_cluster.py``), the ``@ray.task`` decorator is used to define a task that will be executed on the Ray cluster:
 
-This will involve 2 steps -
+.. important::
+   **Set the Ray Dashboard URL connection parameter or RAY_ADDRESS on your airflow worker to connect to your cluster**
 
-Create a YAML file defining your Ray cluster configuration.
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-.. note::
+.. literalinclude:: ../../example_dags/ray_taskflow_example_existing_cluster.py
+   :language: python
+   :linenos:
+
+
+Ray Cluster Sample Spec (YAML)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+.. important::
     ``spec.headGroupSpec.serviceType`` must be a 'LoadBalancer' to spin a service that exposes your dashboard externally
 
+Save this file in a location accessible to your Airflow installation, and reference it in your DAG code.
+
 .. literalinclude:: ../../example_dags/scripts/ray.yaml
+   :language: yaml
 
-Save this file in a location accessible to your Airflow installation, and reference it in your DAG code.
 
+Example 2: Using @ray.task for job lifecycle
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-Sample DAG (``setup_teardown.py``):
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The below example showcases how to use the ``@ray.task`` decorator to manage the full lifecycle of a Ray cluster: setup, job execution, and teardown.
 
-.. literalinclude:: ../../example_dags/setup-teardown.py
+This approach is ideal for jobs that require a dedicated, short-lived cluster, optimizing resource usage by cleaning up after task completion.
+
+.. literalinclude:: ../../example_dags/ray_taskflow_example.py
    :language: python
    :linenos:
 
-Scenario 2: Using an existing Ray cluster
------------------------------------------
 
-If you already have a Ray cluster set up, you can use the ``SubmitRayJob`` operator or ``task.ray()`` decorator to submit jobs directly.
+Example 3: Using SubmitRayJob operator for job lifecycle
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-In the example below (``ray_taskflow_example.py``), the ``@task.ray`` decorator is used to define a task that will be executed on the Ray cluster:
+This example demonstrates how to use the ``SubmitRayJob`` operator to manage the full lifecycle of a Ray cluster and job execution.
```

Comment:
What happens if there are two DAGs using `SubmitRayJob` and the same cluster conn_id, and one of them tries to tear down the Ray cluster while the other is still running a job?

Reply:
In that scenario, they will need to use different Ray cluster specs yaml files. We will just need the cluster name to be different. They can use the same conn_id if required.

If they both wish to use the same cluster, then, they would use the SetupRayCluster operator to spin up a cluster and share the dashboard URL to submit jobs w/o spin/down cluster.